sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobClientBase.java

```diff
@@ -261,23 +262,71 @@ public final BlobInputStream openInputStream(BlobRange range, BlobRequestConditi
      */
     public BlobInputStream openInputStream(BlobInputStreamOptions options) {
         options = options == null ? new BlobInputStreamOptions() : options;
+        ConsistentReadControl consistentReadControl = options.getConsistentReadControl() == null ? ConsistentReadControl.ETAG
+            : options.getConsistentReadControl();
 
-        BlobProperties properties = getProperties();
+        BlobProperties properties = getPropertiesWithResponse(options.getRequestConditions(), null, null).getValue();
+        String eTag = properties.getETag();
+        String versionId = properties.getVersionId();
 
         BlobRange range = options.getRange() == null ? new BlobRange(0) : options.getRange();
         int chunkSize = options.getBlockSize() == null ? 4 * Constants.MB : options.getBlockSize();
 
         BlobRequestConditions requestConditions = options.getRequestConditions() == null
             ? new BlobRequestConditions() : options.getRequestConditions();
-        // Target the user specified version by default. If not provided, target the latest version.
-        if (requestConditions.getIfMatch() == null) {
-            requestConditions.setIfMatch(properties.getETag());
+        BlobAsyncClientBase client = this.client;
+
+        switch (consistentReadControl) {
+            case NONE:
+                if (requestConditions.getIfMatch() != null) {
+                    throw logger.logExceptionAsError(generateConsistentReadControlException("requestConditions.ifMatch",
+                        ConsistentReadControl.NONE.toString(), ConsistentReadControl.ETAG.toString()));
+                }
+                if (this.client.getVersionId() != null) {
+                    throw logger.logExceptionAsError(generateConsistentReadControlException("client.versionId",
+                        ConsistentReadControl.NONE.toString(), ConsistentReadControl.VERSION_ID.toString()));
+                }
+                break;
+            case ETAG:
+                if (this.client.getVersionId() != null) {
+                    throw logger.logExceptionAsError(generateConsistentReadControlException("client.versionId",
+                        ConsistentReadControl.ETAG.toString(), ConsistentReadControl.VERSION_ID.toString()));
+                }
```

Comment:
If someone is using both versionid and ifmatch condition this is going to break them.
Maybe we shouldn't default to "ETAG" few lines above but rather apply this switch block only if consistency controll was explicitly set. Otherwise use old logic?

Reply:
I think Rick and I already talked about this - there really is no reason to be using both version and ifMatch conditions (since theyre basically duplicating work).